hoomd/mpcd/CollisionMethod.cc

```diff
@@ -176,12 +136,15 @@ void mpcd::detail::export_CollisionMethod(pybind11::module& m)
         m,
         "CollisionMethod")
         .def(pybind11::init<std::shared_ptr<SystemDefinition>, uint64_t, uint64_t, int>())
-        .def("enableGridShifting", &mpcd::CollisionMethod::enableGridShifting)
+        .def_property_readonly("embed",
+                               [](const std::shared_ptr<mpcd::CollisionMethod> method)
+                               {
+                                   auto group = method->getEmbeddedGroup();
+                                   return (group) ? group->getFilter()
+                                                  : std::shared_ptr<hoomd::ParticleFilter>();
+                               })
         .def("setEmbeddedGroup", &mpcd::CollisionMethod::setEmbeddedGroup)
-        .def("setPeriod", &mpcd::CollisionMethod::setPeriod)
-        .def_property("instance",
-                      &mpcd::CollisionMethod::getInstance,
-                      &mpcd::CollisionMethod::setInstance);
+        .def_property_readonly("every", &mpcd::CollisionMethod::getPeriod);
```

Comment:
I switched to `every` because `period` wasn't used elsewhere anymore, but this is not a general `trigger` either. Typically when people write up these methods they say something like "a collision was performed every 5 steps" so I thought that fit. I can switch back to `period` if you prefer though!

Reply:
I would also prefer `period`. I find class attributes that are not nouns awkward in general.